appserver/handle_topology.go

```diff
@@ -0,0 +1,531 @@
+package appserver
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/gorilla/websocket"
+	deploymentconfigv1 "github.com/openshift/api/apps/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/redhat-developer/app-service/appserver/topology"
+	"github.com/redhat-developer/app-service/kubeclient"
+	"github.com/redhat-developer/app-service/watcher"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var k8log = logf.Log
+
+type nodeMeta struct {
+	ID          string
+	Name        string
+	Type        string
+	Kind        string
+	Value       interface{}
+	Labels      map[string]string
+	Annotations map[string]string
+}
+
+type innerData struct {
+	nm nodeMeta
+	nd topology.NodeData
+}
+
+type nodesMap struct {
+	nodes map[string]innerData
+}
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// HandleTopology returns the handler function for the /status endpoint.
+func (srv *AppServer) HandleTopology() http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Get the namespace and convert the connection to a web stocket.
+		namespace := "default"
```

Comment:
You can then get the namespace as a query parameter like this `r.FormValue("namespace")`

Reply:
updated here: 25c2523b0b128d31418e6068c32c85940b3c1fb8